datalad/distribution/siblings.py

```diff
@@ -584,7 +584,21 @@ def _configure_remote(
                 # that git-annex will use too
                 remote_url = repo.config.get(f'remote.{name}.url')
                 ri = RI(remote_url)
-                if isinstance(ri, URL) and ri.scheme in ('http', 'https'):
+                if not (isinstance(ri, URL) and ri.scheme in ('http', 'https')):
+                    yield dict(
+                        status='impossible',
+                        message='cannot configure as a common data source, '
+                                'URL protocol is not http or https',
+                        **result_props)
+                elif f'remote.{name}.annex-uuid' not in repo.config:
+                    yield dict(
+                        status='impossible',
```

Comment:
See general comment on this approach.

Reply:
It could turn out to work later, I think. Even in the case you are targeting: Just because our last knowledge of that GIN repo is, that it is empty and has no annex-uuid, it could get one and that doesn't necessarily have to happen via THIS clone.

Reply:
Would the warning be just `warnings.Warn(...)` or is there a dedicated method?

Reply:
There should be logger instance in that file (usually called `lgr`). It's then `lgr.warn()`.